react/OrderQueue.tsx

```diff
@@ -11,11 +11,15 @@ import React, {
 
 import { QueueStatus } from './constants'
 import { TaskQueue } from './modules/TaskQueue'
+import { CancellablePromiseLike } from './modules/SequentialTaskQueue'
 
 type ListenFunction = (event: QueueStatus, callback: () => any) => () => void
 
 interface Context {
-  enqueue: (task: () => Promise<any>, id?: string) => PromiseLike<void>
+  enqueue: <T extends any>(
```

Comment:
Why not just `<T>`? Is there a difference?

Reply:
yeah, since this is a `tsx` file, the parser can't distinguish this from a component that has the name `T`, so you need to add the `extends any` to hint the compiler that this is a generic parameter of the function

Reply:
Ah, nice, TIL